File: Resource/51900286-51900712-51900718-51900699/source demo/Window Form/51900286-51900712-51900718-51900699/RestaurentDTO/PhieuNhap.cs
using System;
using System.Data;

namespace RestaurentDTO
{
    public class PhieuNhap
    {
        private string _maPhieuNhap;
        private DateTime _ngayNhap;
        private double _tongGia;
        private string _ghiChu;
        private string _maNCC;
        private string _maNhanVien;

        public PhieuNhap(string maPhieuNhap, DateTime ngayNhap, double tongGia, string ghiChu, string maNCC, string maNhanVien)
        {
            _maPhieuNhap = maPhieuNhap;
            _ngayNhap = ngayNhap;
            _tongGia = tongGia;
            _ghiChu = ghiChu;
            _maNCC = maNCC;
            _maNhanVien = maNhanVien;
        }

        public PhieuNhap(DataRow dataRow)
        {
            _maPhieuNhap = dataRow["MaPhieuNhap"].ToString();
            _ngayNhap = (DateTime)dataRow["NgayNhapHang"];
            _tongGia = (double)dataRow["TongGiaNhap"];
            _ghiChu = dataRow["GhiChu"].ToString();
            _maNCC = dataRow["MaNhaCungCap"].ToString();
            _maNhanVien = dataRow["MaNhanVien"].ToString();
        }

        public string MaPhieuNhap { get => _maPhieuNhap; set => _maPhieuNhap = value; }
        public DateTime NgayNhapHang { get => _ngayNhap; set => _ngayNhap = value; }
        public double TongGiaNhap { get => _tongGia; set => _tongGia = value; }
        public string GhiChu { get => _ghiChu; set => _ghiChu = value; }
        public string MaNCC { get => _maNCC; set => _maNCC = value; }
        public string MaNhanVien { get => _maNhanVien; set => _maNhanVien = value; }
    }
}

File: Dev/51900718/src/main/java/tdtu/advanced/java/thinh68/models/PhieuNhap.java
package tdtu.advanced.java.thinh68.models;

import java.sql.Timestamp;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name = "phieu_nhap")
public class PhieuNhap {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ma_phieu_nhap")
	private Long maPhieuNhap;

	@Column(name = "ngay_nhap_hang")
	private Timestamp ngayNhapHang = new Timestamp(System.currentTimeMillis());

	@Column(name = "tong_gia_nhap")
	private double tongGiaNhap;

	@Column(name = "ghi_chu")
	private String ghiChu;

	@OneToMany(mappedBy = "phieuNhap")
	private List<ChiTietPhieuNhap> phieuNhap_ChiTietPhieuNhaps;

	@ManyToOne
	@JoinColumn(name = "phieu_nhap_nha_cung_cap_foreign_key", referencedColumnName = "ma_nha_cung_cap", nullable = false)
	private NhaCungCap nhaCungCap;

	@ManyToOne
	@JoinColumn(name = "phieu_nhap_nhan_vien_foreign_key", referencedColumnName = "ma_nhan_vien", nullable = false)
	private NhanVien nhanVien;
}